pkg/agent/monitortool/latency_store.go

```diff
@@ -233,3 +234,34 @@ func (s *LatencyStore) DeleteStaleNodeIPs() {
 		}
 	}
 }
+
+// ConvertList converts the latency store to a list of NodeIPLatencyEntry.
+func (l *LatencyStore) ConvertList() []stv1aplpha1.NodeIPLatencyEntry {
+	l.mutex.RLock()
+	defer l.mutex.RUnlock()
+
+	entries := make([]stv1aplpha1.NodeIPLatencyEntry, 0, len(l.nodeIPLatencyMap))
+	for nodeIP, entry := range l.nodeIPLatencyMap {
+		// Find the Node name for the given Node IP
```

Comment:
If we need this, why don't keep another map in latency_store for IP <-> nodeName mapping? I assume this is invoked every time latency is reported, and worst case for traversing thru the entire `nodeTargetIPsMap` is basically guaranteed each time

Reply:
Yes, but maintaining the map of nodename -> nodeip also requires some overhead, and every time a node is added and deleted, you need to re-traverse this map of nodename -> nodeip and update it.